### PR TITLE
feat: adapted component IcSendAmout to use the new common component SendInputAmount

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

The input component for the amount of IC tokens should use the same common component of its ETH counterparty: `SendInputAmount`.

# Changes

- Adapted `IcSendAmount` to have the proper standard validation function.
- Created a function to pass to `SendInputAmount` to calculate the max amount possible.
- Since the fee for IC and ICRC tokens is fixed, there is no need to dynamically refresh the amount nor the fee provided to the input.

# Note

Unrelated to this PR, I included a comment on unused variable `triggerCalculateMax` in `SendInputAmount` and disabled the Lint rule _no-unused-vars_ only for that line.

### CODE REVIEW

File: src/frontend/src/lib/components/send/SendInputAmount.svelte [hunk 1]
```diff
@@ -22,6 +22,8 @@
 		amount = calculateMax?.();
 	};
 
+	// This function is used for test scopes to forcibly trigger the onMax function.
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/send/SendInputAmount.svelte:25]: The function `triggerCalculateMax` is used in `SendAmount` component.

File: src/frontend/src/icp/components/send/IcSendAmount.svelte [hunk 2]
```diff
@@ -27,85 +31,70 @@
 	let fee: bigint | undefined;
 	$: fee = ($token as IcToken).fee;
 
-	const validate = () => {
-		if (invalidAmount(amount)) {
-			amountError = undefined;
-			return;
-		}
-
+	$: customValidate = (userAmount: BigNumber) => {
```
Comment (peterpeterparker) [inline src/frontend/src/icp/components/send/IcSendAmount.svelte:34]: Should we provide the return type here? If yes, in another PR given that it's the same omission in `SendAmount.svelte`

```
$: customValidate = (userAmount: BigNumber): Error | undefined => {
```

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks.

Please just have a look at the comment before merging the PR.

File: src/frontend/src/lib/components/send/SendInputAmount.svelte (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/send/SendInputAmount.svelte:25]: will remove the comment

File: src/frontend/src/icp/components/send/IcSendAmount.svelte (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp/components/send/IcSendAmount.svelte:34]: right! will do

**Comment by peterpeterparker** [comment]
FYI I merged the PR assuming you just did not hit the "merge" button @AntonioVentilii-DFINITY.

**Pull request merged**